instance/static/js/src/instance.js

```diff
@@ -122,12 +122,37 @@ app.controller("Index", ['$scope', 'Restangular', 'OpenCraftAPI', '$q',
             } else if(message.data.type === 'instance_log') {
                 if($scope.selected.instance && $scope.selected.instance.id === message.data.instance_id) {
                     $scope.$apply(function(){
-                        $scope.selected.instance.log_text += message.data.log_entry + '\n';
+                        $scope.selected.instance.log_entries.push(message.data.log_entry);
                     });
                 }
             }
         };
 
+        $scope.error_log = function() {
+            if (!$scope.selected.instance) {
+                return;
+            }
+            var log_entries = $scope.selected.instance.log_entries;
+            var error = [];
+            var counting = false;
+            for (var i = log_entries.length; i > 0; i--) {
+                if (!/ERROR \|/.test(log_entries[i - 1])) {
```

Comment:
Rather than using regex here, now that the log lines are a list, it might be worth getting further in that direction and include the log level along with the log message. Then, add it dynamically to each line, similarly to the way it's done when printing out to the shell console: https://github.com/open-craft/opencraft/blob/b58fabf98301bdc4129e4e3222c131f6235eab4b/opencraft/settings.py#L297